File: mall-shop-web/src/main/java/life/eks/xmall/controller/seller/SellerController.java
package life.eks.xmall.controller.seller;

import com.alibaba.dubbo.config.annotation.Reference;
import life.eks.xmall.api.seller.SellerService;
import life.eks.xmall.common.web.Response;
import life.eks.xmall.pojo.Seller;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author Eks OU
 */
@RestController
@RequestMapping("/seller")
public class SellerController {

    @Reference(version = "1.0.0")
    private SellerService sellerService;

    @PostMapping("/register")
    public Response register(@RequestBody Seller seller) {
        try {
            sellerService.save(seller);
            return Response.success();
        } catch (Exception e) {
            return Response.fail();
        }
    }
}

File: mall-shop-web/src/main/resources/resources/js/xmall/controller/sellerController.js
xmall.controller('sellerController', function ($scope, $controller, sellerService) {
    $controller('baseController', {$scope: $scope});

    //商实申请注册
    $scope.register = function () {
        sellerService.register($scope.entity).success(
            function (response) {
                if (response.code === 200) {
                    location.href = "shoplogin.html";
                } else {
                    alert(response.message);
                }
            }
        );
    };

});	
